core/platform/lf_arduino_support.c

```diff
@@ -0,0 +1,84 @@
+/* Arduino Platform API support for the C target of Lingua Franca. */
+
+/*************
+Copyright (c) 2021, The University of California at Berkeley.
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+1. Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***************/
+
+/** Arduino API support for the C target of Lingua Franca.
+ *  
+ *  @author{Anirudh Rengarajan <arengarajan@berkeley.edu>}
+ */
+
+
+#include <time.h>
+#include <errno.h>
+
+#include "lf_arduino_support.h"
+#include "../platform.h"
+#include "Arduino.h"
+
+instant_t ns_to_microsec(instant_t time) {
+    return (time / 1000) - ((time / 1000) % 4);
```

Comment:
What is the `((time / 1000) % 4)` for?  Actually, what is this function for? Shouldn't an `instant_t` always be in microseconds?  At a minimum, comments are needed here.

Reply:
Since fundamental unit of time is defined in microseconds, this function has been removed since don't need to run any conversions anymore.

Reply:
This artifact was from back when we attempted to sync times with the Arduino's internal clock and when we tried to do a nano to micro conversion under the hood, but the solution of changing the time standard turned out to be much cleaner and easier to work with.